src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/SnapshotTracker.kt

```diff
@@ -27,15 +27,22 @@ import java.util.concurrent.atomic.AtomicReferenceArray
 
 
 /**
- * Manages a snapshot of the global static state.
+ * Manages a snapshot of the global static state or manually added *root*-objects.
  *
- * This class only tracks static memory and memory reachable from it,
+ * This class tracks static memory and memory reachable from it,
  * referenced from the test code. So the whole static state is not recorded, but only a subset of that,
  * which is named *snapshot*.
+ *
+ * Also, manual addition of required to restore objects is possible.
  */
 class SnapshotTracker {
     private val trackedObjects = IdentityHashMap<Any, MutableList<MemoryNode>>()
 
+    /**
+     * Stores manually saved root objects, which will be the starting objects during the traversal when restoring snapshot.
+     */
+    private val rootObjects = Collections.newSetFromMap(IdentityHashMap<Any, Boolean>())
```

Comment:
Why do you need separate set for `root` objects? Why not just store them indise `trackedObjects`?

Reply:
true, there is not need to start restoring from `root` objects and from `Class<*>` classes